doc/adr/0013-download-for-incremental-snapshots.md

```diff
@@ -0,0 +1,91 @@
+# 13. Download for incremental snapshots
+
+Date: 2023-04-18
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, the nodes joining the network are required to download the latest state of the network. However, in case of an extended absence, the node needs to have all incremental snapshots present on disk to determine the latest state of the network. This change requires a modification to the download process, as nodes need to fetch all missing incremental snapshots.
+
+## Decision
+
+After careful consideration of the current network state and the challenges of ensuring consistent and accurate data across all nodes, we have decided to modify the download process for joining nodes to fetch all missing incremental snapshots. This decision is based on the understanding that nodes need to have a complete and validated chain of incremental snapshots to determine the state of the network, especially when they have been absent for an extended period.
+
+The modified download process involves several steps to ensure that nodes have a complete and validated chain of incremental snapshots. The first step is to determine the starting point by querying a peer node for the latest snapshot hash. 
+
+```mermaid
+flowchart TD
+  subgraph Determine starting point
+    A[Get latest snapshot from the Joinee]
+  end
+  subgraph Download process
+	B[Get snapshot with hash X=aaaa]
+  end
+  A --> |latest hash: aaaa| B
+```
+
+The node then checks if it has all the incremental snapshots needed to reach the starting point, and if not, it beings downloading missing snapshots recursively from the latest snapshot to the genesis. All validated snapshots are stored in permanent storage, which contains a correct and validated chain of incremental snapshots. Temporary storage may contain potentially invalid or conflicting snapshots.
```

Comment:
By `starting point`, do you mean the genesis snapshot? If so, would it be better to replace it with `genesis`?

Reply:
No, the starting point to the algorithm is the hash that you pass in the parameter (see the first mermaid diagram). You have to have all incremental snapshots since genesis to the starting point, otherwise you need to download missing snapshots.